src/dwd_dynam.jl

```diff
@@ -196,10 +256,17 @@ show(io::IO, vf::DiscreteMachine) = print(
 eltype(::AbstractMachine{T}) where T = T
 
 
-readout(f::DelayMachine, u::AbstractVector, h = nothing, p = nothing, t = 0) = readout(f)(u, h, p, t)
 readout(f::AbstractMachine, u::AbstractVector, p = nothing, t = 0) = readout(f)(u, p, t)
 readout(f::AbstractMachine, u::FinDomFunction, args...) = readout(f, collect(u), args...)
 
+readout(f::DelayMachine, u::AbstractVector, h = nothing, p = nothing, t = 0) = readout(f)(u, h, p, t)
+
+readout(m::Machine{T,I,S}, u::AbstractVector, x::AbstractVector, p=nothing, t=0) where {T, I<:InstantaneousDirectedInterface{T}, S} = 
```

Comment:
There's an issue here since `get_readouts` for non delay machines only passes 4 arguments (`m, states[i], p, t`), so the readouts don't seem to work right when being used in a solver interface, which depends on induced_dyn/readout.

Reply:
There is a problem here where the readout function for `DelayMachines` is underspecified. It's now solved by specifying the type of interfaces for the original `readout` function for `DelayMachines`